src/sinks/prometheus/exporter.rs

```diff
@@ -326,41 +351,51 @@ mod tests {
 mod integration_tests {
     use super::*;
     use crate::{
-        config::SinkContext,
         event::{Metric, MetricValue},
         http::HttpClient,
+        test_util::{random_string, trace_init},
     };
-    use futures::{stream, task::Poll};
+    use chrono::Utc;
     use serde_json::Value;
-    use std::{pin::Pin, task::Context};
-    use tokio::time::Duration;
+    use tokio::{sync::mpsc, time};
+
+    const PROMETHEUS_ADDRESS: &str = "127.0.0.1:9101";
 
     #[tokio::test]
-    async fn prometheus_scrapes_metrics() {
-        crate::test_util::trace_init();
+    async fn prometheus_metrics() {
+        trace_init();
+
+        prometheus_scrapes_metrics().await;
+        time::delay_for(time::Duration::from_millis(500)).await;
+        reset_on_flush_preiod().await;
```

Comment:
This looks like the only call to this function. If you don't anticipate additional uses of it, I suggest integrating it into this test case.